libraries/AP_RCProtocol/AP_RCProtocol_CRSF.h

```diff
@@ -261,19 +264,43 @@ class AP_RCProtocol_CRSF : public AP_RCProtocol_Backend {
         ELRS_RF_MODE_25HZ,
         ELRS_RF_MODE_50HZ,
         ELRS_RF_MODE_100HZ,
+        ELRS_RF_MODE_100HZ_FULL,
         ELRS_RF_MODE_150HZ,
         ELRS_RF_MODE_200HZ,
         ELRS_RF_MODE_250HZ,
+        ELRS_RF_MODE_333HZ_FULL,        
         ELRS_RF_MODE_500HZ,
+        ELRS_RF_MODE_D250HZ,
+        ELRS_RF_MODE_D500HZ,
+        ELRS_RF_MODE_F500HZ,
+        ELRS_RF_MODE_F1000HZ,
+        ELRS_RF_MODE_D50HZ,
         RF_MODE_MAX_MODES,
         RF_MODE_UNKNOWN,
     };
 
+#if AP_OSD_CRSF_EXTENSIONS_ENABLED    
+    // These power levels are valid for both Crossfire and ELRS systems
+    static constexpr uint16_t tx_powers[] = { 0, 10, 25, 100, 500, 1000, 2000, 250, 50 };    
+
+    // Use a struct that contains the full link stats data
+    struct LinkStatus {
+        int16_t rssi = -1;
+        int16_t link_quality = -1;
+        int8_t rf_mode = -1;
+        int16_t tx_power = -1;
```

Comment:
put the extra stuff inside the existing definition with a #define rather than side-by-side